packages/headless/src/features/recent-results/recent-results-actions.ts

```diff
@@ -0,0 +1,58 @@
+import {createAction} from '@reduxjs/toolkit';
+import {ArrayValue, NumberValue, RecordValue} from '@coveo/bueno';
+import {validatePayload} from '../../utils/validate-payload';
+import {Result} from '../../api/search/search/result';
+import {
+  resultPartialDefinition,
+  validateResultPayload,
+} from '../analytics/analytics-utils';
+
+export interface RegisterRecentResultsCreatorPayload {
+  /**
+   * The recent results viewed by the user prior to instantiating the controller.
+   */
+  results: Result[];
+  /**
+   * The maximum number of queries to retain in the list.
+   */
+  maxLength: number;
+}
+
+const registerRecentResultsPayloadDefinition = {
+  results: new ArrayValue({
+    required: true,
+    each: new RecordValue({values: resultPartialDefinition}),
+  }),
+  maxLength: new NumberValue({required: true, min: 1, default: 10}),
+};
+
+/**
+ * Initialize the `recentResults` state.
+ * @param payload (RegisterRecentResultsCreatorPayload) The initial state and options.
+ */
+export const registerRecentResults = createAction(
+  'recentResults/registerRecentResults',
+  (payload: RegisterRecentResultsCreatorPayload) =>
+    validatePayload(payload, registerRecentResultsPayloadDefinition)
+);
+
+/**
+ * Push a recent result to the list.
+ * @param payload (Result) The recently viewed result to push to the list.
+ */
+export const pushRecentResult = createAction(
+  'recentResults/pushRecentResult',
+  (payload: Result) => {
+    validateResultPayload(payload);
```

Comment:
This validator verifies properties that are needed in order to be able to properly log analytics events are present. 

It's not checking that it's a "full actual" result object, just some partial properties.

So while this seems good enough, I think this could in theory be a source of potential bugs, since we are more interested in the property that allows a result object to build back a clickable link for the end user (I assume?).

So we might instead want to have a dedicated validator just for that, that checks for a valid title and click uri (and potentially more properties down the line).

But, for the sake of making sure that a user of the library in JavaScript is passing in something that kind of ressemble a full result object (and not a string, or a number), that function can do the trick.

I let you be the judge.

Reply:
I agree. I understodd it differently as the partial schema being a "good enough" set of values to validate that it's a result. But you're right that it isn't full-proof. I'll define a complete result schema, based on the Result interface and use that.

Reply:
> 'll define a complete result schema, based on the Result interface and use that.

That might be overkill though, and by definition, it's impossible to know exactly what is in a result, because it varies depending on the context (for example, which fields are available).

So a complete validator definition will be... cumbersome, to say the least. It's going to be huge. 

Humm... 🤔 I'm wondering what are the other guys opinion. Maybe just keeping it like this is fine...

Reply:
It would definitely be huge. I would've only bothered with the required attributes.

Reply:
I agree, I'd keep the validation simple and focused on what's needed vs. the effort needed to be comprehensive.

Reply:
Just so we're clear, we're saying we're ok with using the current partial validation?

Reply:
If the current partial validation is validating the relevant properties, then yes 👍